### PR TITLE
Add Views

### PR DESCRIPTION
**Pull request by Odongo006**
# Hello

I hope this message finds you well. I am writing to submit a pull request for review of the project that I have just completed. I have implemented the design based on the provided sneak peek wireframes and focused on ensuring the functionality of the application, with styling being a nice-to-have secondary consideration at this stage.

## Project Requirements

1. Implement the design from the sneak peek wireframes.
2. Focus on making sure everything functions first (styling is secondary).
3. Note: Ignore login logic as we will be using Devise in the upcoming milestone.
4. Analyze the wireframes deeply to identify all necessary elements and user navigation between pages.
5. Utilize the methods created in the Blog App for processing data in models.
6. Populate the database with relevant data to see posts and comments.

The project has been developed with adherence to the above requirements, and I have thoroughly tested the functionality to ensure a smooth user experience. The completed wireframes include:

1. All Users Page Wireframe
2. Single User Page Wireframe
3. All Posts by a Given User Page Wireframe
4. Single Post Page Wireframe

I kindly request that you review the code, functionality, and implementation of the wireframes. Any feedback, suggestions, or improvements would be greatly appreciated to enhance the project.

Thank you in advance for taking the time to review my work. I am eager to receive your feedback and incorporate any necessary improvements. Should you have any questions or require further information, please feel free to reach out to me.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1 +1,36 @@
...
+        <% end %>
+        <p><%= post.text %></p>
+          <div class="count">
+            <p>Comments: <%= post.comments_counter %></p>
+            <p>Likes: <%= post.likes_counter %></p>
+          </div>
+     </div>
+     <% if post.comments_counter > 0 %>
+      <ul class="comments">
+            <% post.comments.each do |comment| %>
```
Comment (Ismailco) [inline app/views/posts/index.html.erb]: - [ ] Kindly make sure to use the `recent_comments` method to render the last 5 comments in the posts index page

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1 +1,15 @@
...
+      </div>
+      <div class="user-details">
+        <%= link_to user_url(user), class: 'name' do %>
+          <%= user.name %>
+        <% end %>
+        <p class="posts-no">
+          <%= "Number of posts: #{user.posts_counter}" %></p>
+      </div>
+    </div>
+  <% end %>
```
Comment (Ismailco) [inline app/views/users/index.html.erb:14]: - [ ] Kindly make sure the views of the app look like the provided wireframe

|Your website|Provided Wirframes|
|-|-|
|Screenshot_20230629_003530||

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,32 @@
...
+        <h4><%= post.text %></h4>
+          <div class="count">
+            <p>Comments: <%= post.comments_counter %></p>
+            <p>Likes: <%= post.likes_counter %></p>
+          </div>
+     </div>
+<% end %>
+<%= link_to user_posts_url(@user.id), class: 'btn-link' do %>
+  <button>See all posts</button>
+<% end %>
```
Comment (Ismailco) [inline app/views/users/show.html.erb:32]: - [ ] Kindly make sure the views of the app look like the provided wireframe

|Your website|Provided Wirframes|
|-|-|
|Screenshot_20230629_003537||

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,36 @@
...
+                <li><%= comment.author.name %>: <%= comment.text %></li>
+            <% end %>
+      </ul>
+     <%end%>
+  <% end %>
+  <div class="btn-link">
+    <button>Pagination</button>
+  </div>
```
Comment (Ismailco) [inline app/views/posts/index.html.erb]: - [ ] Kindly make sure the views of the app look like the provided wireframe

|Your website|Provided Wirframes|
|-|-|
|Screenshot_20230629_003543||

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -1 +1,21 @@
...
+</div>
+
+<% if @post.comments_counter > 0 %>
+  <div class="comments">
+    <% @comments.each do |comment| %>
+      <li><%= comment.author.name %>:
+      <%= comment.text %></li>
+    <% end %>
+  </div>
+<%end%>
```
Comment (Ismailco) [inline app/views/posts/show.html.erb:21]: - [ ] Kindly make sure the views of the app look like the provided wireframe

|Your website|Provided Wirframes|
|-|-|
|Screenshot_20230629_003549||

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Odongo006, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb [hunk 6]
```diff
@@ -1 +1,36 @@
...
+          <div class="count">
+            <p>Comments: <%= post.comments_counter %></p>
+            <p>Likes: <%= post.likes_counter %></p>
+          </div>
+     </div>
+     <% if post.comments_counter > 0 %>
+      <ul class="comments">
+            <% post.comments.each do |comment| %>
+                <li><%= comment.author.name %>: <%= comment.text %></li>
+            <% end %>
```
Comment (TadesseDev) [inline app/views/posts/index.html.erb]: - [ ] Dear @Odongo006, Great job adding some of the changes, but as suggested by the previous reviewer, we need to use the [recent_comments](https://github.com/Odongo006/Rails_blog_App/blob/Views/app/models/post.rb) method that is defined for your posts inside the `Post` controller. This way we will render only the most recent comments.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @Odongo006 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🎉🍾**
Hello @Odongo006 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**